src/index.ts

```diff
@@ -213,15 +220,21 @@ async function addResolverOutput(
 	additionalTemplateVars?: Dictionary<string>,
 ): Promise<void> {
 	// Now read the file, allow the resolver to process it, and return it
-	const extraFiles = await resolver.resolve(
+	const dockerfile = await resolver.resolve(
 		bundle,
 		specifiedDockerfilePath,
 		additionalTemplateVars,
 	);
 
-	for (const file of extraFiles) {
-		pack.entry({ name: file.name, size: file.size }, file.contents);
+	let content = dockerfile.contents.toString();
+	const newContent = await bundle.callDockerfileHook(content);
+	if (newContent != null) {
+		// newContent is typed void | string, to avoid a
```

Comment:
Why is it typed `void | string`? That makes no sense and I don't think it's a breaking change to change the void to `undefined` or `null | undefined` - in fact it probably fixes a bunch of things downstream like this